### PR TITLE
bug: Datafusion doesn't respect case sensitive table references

### PR DESCRIPTION
**Pull request by xhwhis**
## Which issue does this PR close?

Closes #.

## Rationale for this change

Table alias will process uppercase alias into lowercase. This PR will fix it.

## What changes are included in this PR?

`TableReference::parse_str` method converts uppercase to lowercase. The alias here has already undergone `normalize`, so we can directly create a TableReference as a parameter.

## Are these changes tested?

Yes

## Are there any user-facing changes?

No

### CODE REVIEW

**Review by comphead — COMMENTED** [review]
Thanks @xhwhis for your contribution
Would you mind adding more details on reason why this change is need, and would be great to cover the change with unit tests

**Comment by xhwhis** [comment]
> 谢谢@xhwhis对于您的贡献， 您介意添加更多详细信息来说明为什么需要进行此更改，并且很高兴通过单元测试来涵盖此更改

```
DataFusion CLI v34.0.0
❯ CREATE TABLE IF NOT EXISTS valuetable(c1 INT, c2 VARCHAR) AS VALUES(1,'HELLO'),(12,'DATAFUSION');
0 rows in set. Query took 0.066 seconds.

❯ SELECT "TEST".c1 from valuetable as "TEST";
Schema error: No field named test.c1. Valid fields are test.c1, test.c2.
```

**Comment by xhwhis** [comment]
> Thanks @xhwhis for your contribution Would you mind adding more details on reason why this change is need, and would be great to cover the change with unit tests

You can use `datafusion-cli` to observe the uppercase bug in table alias.
```
DataFusion CLI v34.0.0
❯ CREATE TABLE IF NOT EXISTS valuetable(c1 INT, c2 VARCHAR) AS VALUES(1,'HELLO'),(12,'DATAFUSION');
0 rows in set. Query took 0.066 seconds.

❯ SELECT "TEST".c1 from valuetable as "TEST";
Schema error: No field named test.c1. Valid fields are test.c1, test.c2.
```
I believe this PR have a minimal impact. I have added unit tests to cover the changes. Please review.

**Review by comphead — COMMENTED** [review]
I'm able to reproduce with 

```
❯ select "T".* from (select 1 a) as "T";
Error during planning: Invalid qualifier T
❯ select t.* from (select 1 a) as "T";
+---+
| a |
+---+
| 1 |
+---+
1 row in set. Query took 0.005 seconds.
```

Please include this test into the test case, and would be nice to move tests into `select.slt` sqllogic test file

File: datafusion/sql/tests/sql_integration.rs [hunk 1]
```diff
@@ -4278,6 +4294,23 @@ fn test_table_alias() {
         \n          TableScan: person";
     quick_test(sql, expected);
 
+    let sql = "select * from (\
```
Comment (comphead) [inline datafusion/sql/tests/sql_integration.rs:4297]: I think we getting closer, please move those tests to `select.rs` you may want to use `EXPLAIN` keyword to get the plan

Comment (xhwhis) [reply datafusion/sql/tests/sql_integration.rs]: Done.

File: datafusion/sqllogictest/test_files/cte.slt [hunk 2]
```diff
@@ -19,3 +19,20 @@ query II
 select * from (WITH source AS (select 1 as e) SELECT * FROM source) t1,   (WITH source AS (select 1 as e) SELECT * FROM source) t2
 ----
 1 1
+
+query I
+WITH "T" AS (SELECT 1 a) SELECT * FROM "T"
```
Comment (comphead) [inline datafusion/sqllogictest/test_files/cte.slt:24]:
```suggestion
WITH "T" AS (SELECT 1 a) SELECT "T".* FROM "T"
```

**Review by comphead — COMMENTED** [review]
Thanks @xhwhis we are really close

File: datafusion/sqllogictest/test_files/cte.slt (hunk 2 above)
Comment (xhwhis) [reply datafusion/sqllogictest/test_files/cte.slt]: Done.

**Review by comphead — APPROVED** [review]
lgtm thanks @xhwhis
since it is first time contribution I'm leaving it for sometime to let other people review as well

File: datafusion/sqllogictest/test_files/cte.slt [hunk 3]
```diff
@@ -19,3 +19,20 @@ query II
 select * from (WITH source AS (select 1 as e) SELECT * FROM source) t1,   (WITH source AS (select 1 as e) SELECT * FROM source) t2
 ----
 1 1
+
+query I
```
Comment (comphead) [inline datafusion/sqllogictest/test_files/cte.slt:23]: Please add a test description

File: datafusion/sqllogictest/test_files/cte.slt [hunk 4]
```diff
@@ -19,3 +19,20 @@ query II
...
+WITH "T" AS (SELECT 1 a) SELECT "T".* FROM "T"
+----
+1
+
+query TT
```
Comment (comphead) [inline datafusion/sqllogictest/test_files/cte.slt:28]: ditto

File: datafusion/sqllogictest/test_files/cte.slt (hunk 3 above)
Comment (alamb) [reply datafusion/sqllogictest/test_files/cte.slt:23]:
```suggestion
# Ensure table aliases can be case sensitive
query I
```

File: datafusion/sqllogictest/test_files/cte.slt (hunk 4 above)
Comment (alamb) [inline datafusion/sqllogictest/test_files/cte.slt:28]:
```suggestion
# Ensure table aliases can be case sensitive
query TT
```

**Review by alamb — APPROVED** [review]

File: datafusion/sqllogictest/test_files/cte.slt (hunk 3 above)
Comment (xhwhis) [reply datafusion/sqllogictest/test_files/cte.slt:24]: Done. @alamb @comphead

**Review by alamb — APPROVED** [review]
Thank you @xhwhis  -- this looks great. I plan to merge it after CI passes

**Comment by alamb** [comment]
Thanks again @xhwhis

**Pull request merged**